shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1057,6 +1057,16 @@ void WebContents::AddNewContents(
   v8::HandleScope handle_scope(isolate);
   auto api_web_contents =
       CreateAndTake(isolate, std::move(new_contents), Type::kBrowserWindow);
+
+  // We call RenderFrameCreated here as at this point the empty "about:blank"
+  // render frame has already been created.  If the window never navigates again
+  // RenderFrameCreated won't be called and certain prefs like
+  // "kBackgroundColor" will not be applied.
+  auto* frame = api_web_contents->MainFrame();
+  if (frame) {
+    api_web_contents->RenderFrameCreated(frame);
```

Comment:
It's extremely confusing to have WebContentsObserver methods called from non-WebContents places. Extract the common logic and call it from both here and the observer method.